code/parachain/frame/composable-traits/src/currency.rs

```diff
@@ -142,13 +163,123 @@ impl<
 {
 }
 
-// hack to imitate type alias until it is in stable
-// named with like implying it is`like` is is necessary to be `AssetId`, but may be not enough (if
-// something is `AssetIdLike` than it is not always asset)
+pub trait AssetIdLike = FullCodec + MaxEncodedLen + Copy + Eq + PartialEq + Debug + TypeInfo;
 
-// FIXME(hussein-aitlahcen): this trait already exists in frame_support, named `AssetId`
-pub trait AssetIdLike:
-	FullCodec + MaxEncodedLen + Copy + Eq + PartialEq + Debug + TypeInfo
-{
+#[derive(
+	RuntimeDebug,
+	Copy,
+	Clone,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	Encode,
+	Decode,
+	MaxEncodedLen,
+	TypeInfo,
+)]
+pub struct Rational64 {
+	pub n: u64,
+	pub d: u64,
+}
```

Comment:
This is almost entirely copied from https://paritytech.github.io/substrate/master/src/sp_arithmetic/rational.rs.html#92 - why not just use `Rational128`?

Reply:
(I'm assuming it's due to size requirements - since `Ratio` was previously used, which was a type alias for `FixedU128`?)

Reply:
partially yes. 
Rational128 is not SCALE.

Reply:
Could you use a newtype wrapper around Rational128 instead? Strange that it doesn't implement any of the scale traits.

Reply:
it will take 256 bit. it is not 128 as fixed was, nor we need such precision.  not sure why parity has not u64?

Reply:
They tend to half-way implement a lot of things, it would have been trivial to have Rational8/16/32/64 implemented with a macro. Perhaps we should raise an issue or submit a PR.